pkg/utils/utils.go

```diff
@@ -20,10 +26,27 @@ func FletcherChecksum8(inp string) uint8 {
 }
 
 func ShortHostname() (shortName string, err error) {
-	hostname, err := os.Hostname()
-	if err != nil {
-		panic(err)
+	var hostname string
+
+	if _, err := os.Stat(mdns_hostname_file); err == nil {
+		dat, err := ioutil.ReadFile(mdns_hostname_file)
+		if err != nil {
+			panic(err)
+		}
+		hostname = string(dat[:len(dat)-1])
```

Comment:
What is the character that you are trimming? `\n` ? If so, it is nicer to use https://golang.org/pkg/strings/#TrimSpace